.github/actions/poetry-install/action.yml

```diff
@@ -0,0 +1,42 @@
+name: Poetry Install
+description: Run `poetry install` with optional artifact and metadata caching
+
+inputs:
+  args:
+    description: Arguments for `poetry install`
+  cache:
+    description: Enable transparent Poetry artifact and metadata caching
+    default: 'true'
+
+outputs:
+  cache-hit:
+    description: Whether an exact cache hit occured
+    value: ${{ steps.cache.outputs.cache-hit }}
+
+runs:
+  using: composite
+  steps:
+    - run: printf 'cache-dir=%s\n' "$(poetry config cache-dir)" >> $GITHUB_OUTPUT
+      id: poetry-config
+      shell: bash
+
+    - uses: actions/cache@v4
```

Comment:
Can't we move this to bootstrap? Because we really do not want caches changing during the execution of the job right? And the call to cache action every step is rather overkill.

Reply:
We can't do this in `bootstrap-poetry`, because composite actions trigger post-run actions when they complete. In other words, the cache would never cache because `poetry install` hasn't happened yet.

With the new `poetry-install` composite action, we actually finalize the cache after `poetry install` and before proceeding with the rest of the job.

I'm not sure what you mean about excessive; we only perform this step when `poetry install` happens.